feat: support an include or exclude list of metadata when building a manifest from an org (#1247)
salesforcecli/plugin-deploy-retrieve: messages/manifest.generate.md

To include multiple metadata components, either set multiple --metadata <name> flags or a single --metadata flag with multiple names separated by spaces. Enclose names that contain spaces in one set of double quotes. The same syntax applies to --include-packages and --source-dir.

> To build a manifest from the metadata in an org use the --from-org flag optionally combining it with the --metadata flag to only include certain metadata types, or the --exclude-metadata flag to exclude certain metadata types. When building a manifest from an org, the command makes many API calls concurrently to discover the metadata that exists in the org. To limit the number of concurrent requests use the `SF_LIST_METADATA_BATCH_SIZE` environment variable and set it to a size that works best for your org and environment. If you are experiencing timeouts and/or inconsistent manifest contents then setting this environment variable should improve accuracy. However, the command will take longer to run since it sends fewer requests at a time.

Reviewer: ```suggestion
To build a manifest from the metadata in an org, use the --from-org flag. You can combine --from-org with the --metadata flag to include only certain metadata types, or with the --excluded-metadata flag to exclude certain metadata types. When building a manifest from an org, the command makes many concurrent API calls to discover the metadata that exists in the org. To limit the number of concurrent requests, use the `SF_LIST_METADATA_BATCH_SIZE` environment variable and set it to a size that works best for your org and environment. If you experience timeouts or inconsistent manifest contents, then setting this environment variable can improve accuracy. However, the command takes longer to run because it sends fewer requests at a time.
```

  $ <%= config.bin %> <%= command.id %> --from-org test@myorg.com --include-packages unlocked

- Create a manifest from specific metadata types in an org:

  $ <%= config.bin %> <%= command.id %> --from-org test@myorg.com --metadata ApexClass,CustomObject,CustomLabels

- Create a manifest from all metadata components in an org excluding specific metadata types:

>   $ <%= config.bin %> <%= command.id %> --from-org test@myorg.com --exclude-metadata StandardValueSet

Reviewer: ```suggestion
  $ <%= config.bin %> <%= command.id %> --from-org test@myorg.com --excluded-metadata StandardValueSet
```

Reviewer: Assuming we change the name of the flag as I suggest, of course.

# flags.include-packages.summary

Package types (managed, unlocked) whose metadata is included in the manifest; by default, metadata in managed and unlocked packages is excluded. Metadata in unmanaged packages is always included.

# flags.exclude-metadata.summary

> Metadata types (types only; not names) to exclude when building a manifest from an org.

Reviewer: ```suggestion
Metadata types to exclude when building a manifest from an org. Specify the name of the type, not the name of a specific component. 
```
